File: rockpaperscissors-api/internal/domain/websocket.go
package domain

type WebsocketPayload struct {
	Nickname string                  `json:"nickname,omitempty"`
	GameID   string                  `json:"gameId,omitempty"`
	Game     Game                    `json:"game"`
	Value    Move                    `json:"value,omitempty"`
	Command  WebsocketPayloadCommand `json:"command,omitempty"`
	Message  string                  `json:"message"`
}

type WebsocketPayloadCommand string

const (
	WebsocketPayloadCommandConnect WebsocketPayloadCommand = "connect"
	WebsocketPayloadCommandPlay    WebsocketPayloadCommand = "play"
	WebsocketPayloadCommandOK      WebsocketPayloadCommand = "ok"
	WebsocketPayloadCommandError   WebsocketPayloadCommand = "error"
	WebsocketPayloadCommandClose   WebsocketPayloadCommand = "close"
	WebsocketPayloadCommandPause   WebsocketPayloadCommand = "pause"
	WebsocketPayloadCommandUnpause WebsocketPayloadCommand = "unpause"
)

File: rockpaperscissors-web/src/types/websocket.ts
import { Game as GameType } from "./game";

/**
 * Websocket payload
 */
export interface Payload {
  nickname: string;
  gameId: string;
  game: GameType;
  // value: number;
  command: Command;
  message: string;
}

/**
 * Command to send to the websocket server so it can manage the game status
 */
export type Command = "connect" | "play" | "ok" | "error" | "close";
